### PR TITLE
Closes #3 - Adds Interactions and API

### LINKED ISSUE #3: Add Interaction API

### PR DESCRIPTION
**Pull request by cbetta**
Next up going to work on adding this to herder

Looks more complex than it is cause #21 is still open

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,9 +1,25 @@
...
-      end
+def create_resource_for model, handled = []
+  # we dont want to loop the recursion
+  handled << model
+
+  # we don't need the new and edit form
+  resources model, except: [:new, :edit] do
+    # automagically create nested routes for their direct associations,
+    # used for active resource relations
+    model.to_s.capitalize.singularize.constantize.reflect_on_all_associations.each do |association|
```
Comment (kevinprince) [inline config/routes.rb:9]: This line is epic

Comment (cbetta) [inline config/routes.rb:9]: :trollface:

**Comment by kevinprince** [comment]
:ok:

**Comment by mseckington** [comment]
:ok:

**Pull request merged**